### PR TITLE
SAMBA - filter by group option

### PR DESCRIPTION
**Pull request by theCalcaholic**
Implements #1047

At this point this is a quickly thrown together (untested!) implementation (10min effort :P ).
I'll iron it out in the coming days if Nacho likes the approach.

### CODE REVIEW

File: etc/ncp-config.d/samba.cfg [hunk 1]
```diff
@@ -17,6 +17,11 @@
       "name": "Password",
       "value": "ownyourbits",
       "type": "password"
+    },
+    {
+      "id": "FILTER_BY_GROUP",
+      "name": "Filter by NC group:",
+      "value": ""
```
Comment (nachoparker) [inline etc/ncp-config.d/samba.cfg]: maybe a suggest value `optional_group` and maybe some explanation in the description, but in general looks good to me.

Comment (theCalcaholic) [reply etc/ncp-config.d/samba.cfg]: Sure. Is `OPTIONAL_GROUP_FILTER` okay?

Comment (nachoparker) [reply etc/ncp-config.d/samba.cfg]: sounds good to me

Comment (theCalcaholic) [reply etc/ncp-config.d/samba.cfg]: Ah, I misunderstood you earlier. I've updated the description and added a suggest value

Comment (theCalcaholic) [reply etc/ncp-config.d/samba.cfg]: @nachoparker Let me know if my recent change implements what you had in mind (once you get to it - there's no hurry).

File: bin/ncp/NETWORKING/samba.sh [hunk 2]
```diff
@@ -68,6 +68,14 @@ EOF
   done < <( ls -d "$DATADIR"/*/files )
 
   for user in ${USERS[@]}; do
+    # Exclude users not matching group filter (if enabled)
+    if [[ -n "$FILTER_BY_GROUP" ]] \
+    && [[ -z "$(ncc user:info "$user" --output=json | jq ".groups[] | select( . == \"${FILTER_BY_GROUP}\" )")" ]]
+    then
+      echo "Omitting user $user (not in group ${FILTER_BY_GROUP}...";
```
Comment (nachoparker) [inline bin/ncp/NETWORKING/samba.sh]:
```suggestion
      echo "Omitting user $user (not in group ${FILTER_BY_GROUP})...";
```

File: etc/ncp-config.d/samba.cfg [hunk 3]
```diff
@@ -17,6 +17,12 @@
       "name": "Password",
       "value": "ownyourbits",
       "type": "password"
+    },
+    {
+      "id": "FILTER_BY_GROUP",
+      "name": "Filter by NC group:",
```
Comment (nachoparker) [inline etc/ncp-config.d/samba.cfg]:
```suggestion
      "name": "Filter by NC group",
```

**Comment by nachoparker** [comment]
merged to `devel`. Is it tested by you? good to merge to `master`?

**Comment by theCalcaholic** [comment]
I did not test it yet, because I wanted to hear your opinion on the general idea first!
I'll try to do that tomorrow (I'm on a tight schedule this week, because it's my wife's birthday :P)

**Pull request merged**